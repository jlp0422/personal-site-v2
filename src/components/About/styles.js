import styled from '@emotion/styled'

export const Copy = styled.p`
  font-size: 2rem;
  padding: 0.5rem 0;
`

export const ContactList = styled.div`
  > div {
    font-family: ${({ theme }) => theme.fonts.primary};
    font-size: 1.8rem;
  }
  span,
  a {
    font-size: 1.8rem;
  }
  a {
    cursor: pointer;
  }
`

export const ContentWrapper = styled.div`
  display: grid;
  grid-template-columns: 1fr 350px;
  grid-column-gap: 3%;
  @media only screen and (max-width: 800px) {
    grid-template-columns: 1fr 250px;
  }
  @media only screen and (max-width: 640px) {
    display: block;
  }
`

export const ImageWrapper = styled.div`
  @media only screen and (max-width: 640px) {
    width: 300px;
    margin: 2rem auto;
  }
`
