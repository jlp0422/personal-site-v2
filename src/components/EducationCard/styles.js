import styled from '@emotion/styled'
import {
  CardFlipBack,
  CardFlipContainer,
  CardFlipInner,
  CardShared,
  ExpInfo,
  ExpTitle
} from '../shared'

export const Container = styled(CardFlipContainer)`
  text-align: center;
  @media only screen and (max-width: 640px) {
    height: 250px;
  }
`

export const InnerContainer = styled(CardFlipInner)`
  box-shadow: ${({ theme }) => theme.palette.boxShadow.dark};
`

export const CardFront = styled(CardShared)`
  background-color: ${({ theme }) => theme.palette.black.darkest};
`

export const CardBack = styled(CardFlipBack)`
  background-color: ${({ theme }) => theme.palette.white.lighter};
`

export const Title = ExpTitle

export const Info = styled(ExpInfo)`
  width: 90%;
  grid-template-columns: repeat(2, 1fr);
`
