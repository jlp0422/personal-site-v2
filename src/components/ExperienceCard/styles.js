import styled from '@emotion/styled'
import {
  ExpInfo,
  ExpTitle,
  CardFlipContainer,
  CardFlipInner,
  CardFlipBack,
  CardShared
} from '../shared'

export const Container = styled(CardFlipContainer)`
  display: grid;
  text-align: center;
  margin: 2.5rem auto;
  @media only screen and (max-width: 640px) {
    height: 250px;
  }
`

export const InnerContainer = styled(CardFlipInner)`
  box-shadow: ${({ theme }) => theme.palette.boxShadow.light}
`

export const CardFront = styled(CardShared)`
  padding: 1.5rem;
  background-color: ${({ theme }) => theme.palette.black.darkest};
`

export const CardBack = styled(CardFlipBack)`
  padding: 1.5rem;
  background-color: ${({ theme }) => theme.palette.white.lighter}
`

export const Title = ExpTitle

export const Info = styled(ExpInfo)`
  width: 100%;
  align-items: center;
  @media only screen and (max-width: 640px) {
    grid-template-columns: 1fr;
  }
`

export const LogoContainer = styled.div`
  display: block;
  visibility: visible;
  width: 60px;
  margin: 1rem auto 0;
  @media only screen and (max-width: 640px) {
    display: none;
    visibility: hidden;
  }
`
