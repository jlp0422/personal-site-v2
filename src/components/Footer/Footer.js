import React from "react"

const Footer = () => (
  <footer>
    &copy; {new Date().getFullYear()} Jeremy L. Philipson
  </footer>
)

export default Footer
