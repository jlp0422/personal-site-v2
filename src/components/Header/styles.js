import styled from '@emotion/styled'
import { Link } from 'gatsby'
import { ActiveLink } from '../../components/shared/Styled'

export const StyledHeader = styled.header`
  background: ${({ theme }) => theme.palette.link.secondary};
  margin-bottom: 1.5rem;
`

export const Div = styled.div`
  margin: 0 auto;
  max-width: 1200px;
  padding: 2rem 1.2rem;
  display: grid;
  grid-template-columns: 120px 120px 1fr 120px 120px;
  justify-items: center;
  align-items: center;
  grid-column-gap: 2rem;
  @media only screen and (max-width: 640px) {
    padding: 0.8rem;
    display: flex;
    align-items: center;
    justify-content: center;
  }
`

export const H1 = styled.h1`
  margin: 0;
  text-align: center;
  text-transform: uppercase;
`

export const HeaderLink = styled(Link)`
  letter-spacing: 0.3rem;
  font-family: ${({ theme }) => theme.fonts.header};
  color: white;
  text-decoration: none;
  font-size: 4.5rem;
  @media only screen and (max-width: 800px) {
    visibility: hidden;
    display: none;
  }
`

export const LogoLink = styled(Link)`
  margin: 0;
  visibility: hidden;
  display: none;
  @media only screen and (min-width: 640px) and (max-width: 800px) {
    visibility: visible;
    display: block;
  }
`

export const NavLink = styled(ActiveLink)`
  @media only screen and (max-width: 640px) {
    display: none;
    visibility: hidden;
  }
`

export const MobileLogo = styled.button`
  margin: 0;
  visibility: hidden;
  display: none;
  @media only screen and (max-width: 639px) {
    visibility: visible;
    display: block;
  }
`

export const List = styled.ul`
  padding: 0 0 1rem;
  list-style: none;
  margin-top: 0;
  visibility: hidden;
  display: none;
  @media only screen and (max-width: 640px) {
    visibility: visible;
    display: block;
  }
`
