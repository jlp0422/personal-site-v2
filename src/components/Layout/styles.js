import styled from '@emotion/styled'

export const ContainerDiv = styled.div`
  margin: 0 auto;
  max-width: 1200px;
  padding: 0 1.5rem 1.5rem;
  min-height: 85vh;
  display: grid;
  grid-template-rows: 1fr auto;
`
