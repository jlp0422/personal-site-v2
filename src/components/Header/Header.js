import React from 'react'
import PropTypes from 'prop-types'
import { ActiveLink as NavLink } from '../../components/shared/Styled'
import { StyledHeader, Div, H1, HeaderLink } from './styles'

const Header = ({ siteTitle }) => (
  <StyledHeader>
    <Div>
      <NavLink to="/about">About Me</NavLink>
      <NavLink to="/experience">Experience</NavLink>
      <H1>
        <HeaderLink to="/">{siteTitle}</HeaderLink>
      </H1>
      <NavLink to="/projects">Projects</NavLink>
      <NavLink to="/resume">Resume</NavLink>
    </Div>
  </StyledHeader>
)

Header.propTypes = {
  siteTitle: PropTypes.string
}

Header.defaultProps = {
  siteTitle: ``
}

export default Header
