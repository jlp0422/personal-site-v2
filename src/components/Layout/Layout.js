import { ThemeProvider } from 'emotion-theming'
import { graphql, useStaticQuery } from 'gatsby'
import PropTypes from 'prop-types'
import React from 'react'
import { theme } from '../../helpers'
import Footer from '../Footer'
import GlobalStyle from '../GlobalStyle'
import Header from '../Header'
import { ContainerDiv } from './styles'

export const PureLayout = ({ data, children }) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      <Header siteTitle={data.site.siteMetadata.title} />
      <ContainerDiv>
        <main>{children}</main>
        <Footer />
      </ContainerDiv>
    </ThemeProvider>
  )
}

const Layout = ({ children }) => {
  const data = useStaticQuery(graphql`
    query SiteTitleQuery {
      site {
        siteMetadata {
          title
        }
      }
    }
  `)

  return <PureLayout data={data}>{children}</PureLayout>
}

Layout.propTypes = {
  children: PropTypes.node.isRequired
}

export default Layout
