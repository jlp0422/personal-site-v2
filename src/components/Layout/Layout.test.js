import { ThemeProvider } from 'emotion-theming'
import React from 'react'
import renderer from 'react-test-renderer'
import Footer from '../Footer'
import Header from '../Header'
import { PureLayout as Layout } from './Layout'
import { ContainerDiv } from './styles'

describe('Layout', () => {
  const data = {
    site: {
      siteMetadata: {
        title: 'Gatsby Starter Blog'
      }
    },
  }
  const getOutput = () => renderer.create(<Layout data={data} />)

  it('Renders correctly', () => {
    const tree = getOutput().toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Wraps everything in the theme provider', () => {
    const instance = getOutput().root
    const component = instance.findByType(ThemeProvider)
    expect(component).toBeTruthy()
    expect(component.props).toHaveProperty('theme')
  })

  it('Passes the sietMetaData title to the Header component', () => {
    const instance = getOutput().root
    const component = instance.findByType(Header)
    expect(component).toBeTruthy()
    expect(component.props.siteTitle).toBe('Gatsby Starter Blog')
  })

  it('Wraps the main content and Footer in a ContainerDiv', () => {
    const instance = getOutput().root
    const component = instance.findByType(ContainerDiv)
    const children = component.props.children
    expect(component).toBeTruthy()
    expect(children.length).toBe(2)
    expect(children[0].type).toBe('main')
    expect(children[1].type).toEqual(Footer)
  })
})
