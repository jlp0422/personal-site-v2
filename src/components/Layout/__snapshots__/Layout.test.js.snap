// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Renders correctly 1`] = `
Array [
  <header
    className="css-1t9b64j"
  >
    <div
      className="css-wnxrhc"
    >
      <a
        className="css-1fu3yhm"
        href="/about"
      >
        About
      </a>
      <a
        className="css-1fu3yhm"
        href="/experience"
      >
        Experience
      </a>
      <h1
        className="css-11z6ppj"
      >
        <div
          className="css-1axmddg"
        >
          Gatsby Starter Blog
        </div>
        <div
          className="css-1n7cfic"
        >
          <img
            alt="logo_black"
            height={60}
            src="/static/1f842f35d1f3b3fede85860b885408c2/logo_black.svg"
            title="logo_black"
            width={60}
          />
        </div>
        <button
          className="css-1uq4jnp"
          onClick={[Function]}
        >
          <img
            alt="logo_black"
            height={60}
            src="/static/1f842f35d1f3b3fede85860b885408c2/logo_black.svg"
            title="logo_black"
            width={60}
          />
        </button>
      </h1>
      <a
        className="css-1fu3yhm"
        href="/projects"
      >
        Projects
      </a>
      <a
        className="css-1fu3yhm"
        href="/resume"
      >
        Resume
      </a>
    </div>
  </header>,
  <div
    className="css-tad0pn"
  >
    <main />
    <footer>
      Â© 
      2020
       Jeremy L. Philipson
    </footer>
  </div>,
]
`;
